js/main.js

```diff
@@ -47,15 +57,54 @@
     },
 
     createTheme: function() {
-      var title = prompt('Title');
-      if (!title) {
-        return;
-      }
-      Storage.createTheme(title).then(() => {
-        this.prepareForDisplay();
-      }).catch(function(error) {
-        console.log(error);
+      this.promptNewTheme().then((theme) => {
+        if (!theme.title) {
+          return;
+        }
+
+        theme = ThemeCreator.template(theme);
+
+        Storage.createTheme(theme).then(() => {
+          this.prepareForDisplay();
+        }).catch(function(error) {
+          console.log(error);
+        });
       });
+    },
+
+    promptNewTheme() {
+      AutoTheme.on('palette', this.onPalette);
+      this.dialog.open();
+      this.createDialogDefer = new Defer();
+      return this.createDialogDefer.promise;
+    },
+
+    closeDialog() {
+      this.dialog.close();
+      this.dialogInput.value = '';
+      AutoTheme.off('palette', this.onPalette);
+      AutoTheme.clean();
+      this.onPalette();
+      this.createDialogDefer = null;
+    },
+
+    onPalette() {
+      // no "this" available here
+      AutoTheme.showPalette(Main.autotheme);
```

Comment:
Since generating the palette takes some work, I feel bad just silently discarding the theme if no title was set in this case (L61-62). An easy fix would be, if the title field is empty, once the palette is generated set it to `Auto Theme`